addons/html_editor/static/src/others/embedded_components/backend/caption/caption.js

```diff
@@ -0,0 +1,119 @@
+import {
+    applyObjectPropertyDifference,
+    getEmbeddedProps,
+    StateChangeManager,
+} from "@html_editor/others/embedded_component_utils";
+import { Component, useState, useRef, onMounted } from "@odoo/owl";
+
+export class EmbeddedCaptionComponent extends Component {
+    static template = "html_editor.EmbeddedCaption";
+
+    static props = {
+        id: { type: String },
+        editable: { type: Element },
+        addHistoryStep: { type: Function },
+        undo: { type: Function },
+        redo: { type: Function },
+        focusInput: { type: Boolean },
+        host: { type: Object },
+    };
+
+    setup() {
+        super.setup();
+        this.image = this.props.editable.querySelector(`img[data-caption-id="${this.props.id}"]`);
+        this.state = useState({
+            caption: "",
+            host: this.props.host,
+        });
+        this.captionInput = useRef("captionInput");
+        if (this.props.focusInput) {
+            onMounted(() => {
+                this.captionInput.el.focus();
+            });
+        }
+        // Ensure the state, the attribute and the placeholder are in sync.
+        // We update without adding a history step because it will be added by
+        // the plugin.
+        this.updateCaption(this.image.getAttribute("data-caption"), false);
+        this.observer = new MutationObserver((mutations) => {
+            for (const mutation of mutations) {
+                if (mutation.type === "attributes" && mutation.attributeName === "data-caption") {
+                    this.updateCaption(this.image.getAttribute("data-caption"));
+                }
+            }
+        });
+        this.observer.observe(this.image, { attributes: true });
+    }
+
+    destroy() {
+        this.observer.disconnect();
+    }
+
+    updateCaption(caption = "", addHistoryStep = true) {
```

Comment:
related to the this file comment.
I hate this `addHistoryStep = true` parameter. It is such a in hindsight parameter. I would really like to avoid those kind of stuff if possible.